modules/lib/src/main/resources/assets/js/app/util/WidgetHelper.ts

```diff
@@ -0,0 +1,51 @@
+import {Element as UIElement} from 'lib-admin-ui/dom/Element';
+
+export class WidgetHelper {
+
+    static injectWidgetHtml(html: string, target: UIElement): { scriptElements: HTMLScriptElement[], resultElement: UIElement } {
+        const widgetContainer: HTMLHtmlElement = document.createElement('html');
+        widgetContainer.innerHTML = html;
```

Comment:
Create an Element using `UIElement.fromHtmlElement` as early as possible, then use `Element.setHtml()` to ensure HTML gets escaped.